virtcontainers/qemu.go

```diff
@@ -1341,3 +1431,45 @@ func (q *qemu) getThreadIDs() (*threadIDs, error) {
 	}
 	return &tid, nil
 }
+
+func calcHotplugMemMiBSize(mem uint32, memorySectionSizeMB uint32) (uint32, error) {
+	if memorySectionSizeMB == 0 {
+		return mem, nil
+	}
+
+	// TODO: hot add memory aligned to memory section should be more properly. See https://github.com/kata-containers/runtime/pull/624#issuecomment-419656853
+	return uint32(math.Ceil(float64(mem)/float64(memorySectionSizeMB))) * memorySectionSizeMB, nil
+}
+
+func (q *qemu) resizeVCPUs(reqVCPUs uint32) (currentVCPUs uint32, newVCPUs uint32, err error) {
+
+	currentVCPUs = q.config.NumVCPUs + uint32(len(q.state.HotpluggedVCPUs))
```

Comment:
Is NumVCPUs the default, and hot plugged is what we've added?  Should the default be included in the CPU calculation here, since, AFAIU, it shouldn't be constrained/part of the workload?